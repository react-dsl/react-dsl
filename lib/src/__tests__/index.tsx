import * as React from 'react';
import { create } from 'react-test-renderer';

import { DefaultCache, DslProvider, withDsl } from '..';

const methods = {
    $(a) {
        return this[a];
    },
};

const state = {
    a: 1,
    b: 2,
};

const Foo = ({ text }) => <span>{JSON.stringify(text)}</span>;

describe('DSL', () => {

    test('less than operator', () => {
        const cache = new DefaultCache();
        const ConnectedFoo = withDsl((dsl, props) => ({
            text: dsl.calculate({}, props.text),
        }))(Foo);
        const App = () => (
            <DslProvider cache={cache} methods={methods} prefix={null}>
                <ConnectedFoo text='1 > 2' />
            </DslProvider>
        );
        const component = create(<App />).toJSON();
        expect(component).toMatchSnapshot();
    });

    test('greater than operator', () => {
        const cache = new DefaultCache();
        const ConnectedFoo = withDsl((dsl, props) => ({
            text: dsl.calculate({}, props.text),
        }))(Foo);
        const App = () => (
            <DslProvider cache={cache} methods={methods} prefix={null}>
                <ConnectedFoo text='1 < 2' />
            </DslProvider>
        );
        const component = create(<App />).toJSON();
        expect(component).toMatchSnapshot();
    });

    test('addition operator', () => {
        const cache = new DefaultCache();
        const ConnectedFoo = withDsl((dsl, props) => ({
            text: dsl.calculate({}, props.text),
        }))(Foo);
        const App = () => (
            <DslProvider cache={cache} methods={methods} prefix={null}>
                <ConnectedFoo text='1 + 2' />
            </DslProvider>
        );
        const component = create(<App />).toJSON();
        expect(component).toMatchSnapshot();
    });

    test('methods', () => {
        const cache = new DefaultCache();
        const ConnectedFoo = withDsl((dsl, props) => ({
            text: dsl.calculate(props.state, props.text),
        }))(Foo);
        const App = () => (
            <DslProvider cache={cache} methods={methods} prefix={null}>
                <ConnectedFoo text='$("a") + $("b")' state={state} />
            </DslProvider>
        );
        const component = create(<App />).toJSON();
        expect(component).toMatchSnapshot();
    });
});
