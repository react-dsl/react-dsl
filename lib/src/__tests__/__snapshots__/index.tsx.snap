// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DSL addition operator 1`] = `
<span>
  3
</span>
`;

exports[`DSL greater than operator 1`] = `
<span>
  true
</span>
`;

exports[`DSL less than operator 1`] = `
<span>
  false
</span>
`;

exports[`DSL methods 1`] = `
<span>
  3
</span>
`;
